### ISSUE TITLE
Making a GUI

### ISSUE THREAD

**Issue by GameTec-live** [red-prig/fpPS4#33]
I just want to reach out, if there would be any interest, if i went and made a python gui/frontend...
Is there any interest/would there be a chance that this would be merged if i made one?

**Comment by GameTec-live** [comment]
not really, python is my main language and i know my way around it and with it not replacing anything, just being a more or less independet addition it in my head makes sense? also: ive made multiple guis in python already and it isnt really hard....

**Comment by JJMorty** [comment]
surely since nobody else is doing it and it doesnt affect anything else it would be a good addition even if it's replaced later on with something else like GTK or Avalonia

**Comment by GameTec-live** [comment]
@JJMorty Python/Tkinter would also have the benefit of being crossplatform, AFAIK GTK is linux only?

**Comment by red-prig** [comment]
@GameTec-live Okay, I merge this. (I hope I don't regret it....)

**Comment by GameTec-live** [comment]
i mean, i cant force you to merge it, but thanks... ill start wirking tomorrow, ill leave the issue open incase i have any questions...

**Comment by SlrHui** [comment]
[frofpp4.zip](https://github.com/red-prig/fpPS4/files/10174152/frofpp4.zip)

Hello,
attached is a very very simple gui program for fpPS4.
Source code included. It is programmed in Pascal(Lazarus).
in the ini-file are the 2 paths one for emu and one for apps

Kind regards
SLRHUI

**Comment by red-prig** [comment]
@SlrHui This is a very simple program with several problems, firstly, opening child windows inside a window is not the best idea (I generally have them open at the bottom, probably due to the fixed coordinates when opening the window) besides, I got AV several times ))) And check it out better in the github repository, or, about the pull request to the emulator (preferably in a separate folder) As a prototype for the gui is fine.

**Comment by GameTec-live** [comment]
> 

ok, so ig i dont need to make one then?

**Comment by red-prig** [comment]
@GameTec-live I do not want to add to myself that gui that was posted above, it is very raw, so nvm

**Comment by GameTec-live** [comment]
> I don't see why it's an issue to have a GUI (or several made by different devs) in different repos.
> 
> Personally if they (the GUIs) aren't part of the primary codebase they shouldn't be merged into it, a simple link to them in a README.md would suffice. This way the repo stays clean and other users have the option of using the GUI (or not)

You have a point, but having to download a different app might just throw people of?

**Comment by SlrHui** [comment]
i wrote this gui for myself a few months ago.
it is enough for me. i only made it available in case it is of interest. there will be no new version.

**Comment by GameTec-live** [comment]
[GUI.zip](https://github.com/red-prig/fpPS4/files/10183212/GUI.zip)
not pretty (yet) but should work?

**Comment by red-prig** [comment]
@GameTec-live Yes, it works, but you probably need a list of games instead of choosing one. (Fpps4 autodetect behaves strangely lol) In addition, the application freezes when the game starts, I don’t know if it’s possible to do asynchronous launch of applications and spy on them. PS: Please add some information about yourself to the readme file and probably some information about installing python. PPS: I can add this as a separate folder, but is it really necessary? Or is it still more convenient to leave a link to your repository?

**Comment by GameTec-live** [comment]
@red-prig This was a more or less POC i just threw together... I also already had the idea of a gamebrowser/list where you can just select them (maybe with coverart?)... It freezes when the game starts, because i hand over the control etc to fpps4.exe, so the gui cant update, async calls could work, ill look into this.... Installing python should be easy enough and you can actually package a .py file into an exe... All of the git stuff, eg Readme will come later, rn its just a folder... And you can just add a repository link, altough it more or less being part of the official thing would be my preferred choice and may encrourage people to improve it... Rn i have a friend over, so ill do some improvements in about an hour.... PS: the autodetect is lazily implemented XD it just goes up 2 folders and then searches for a fpps4.exe ;)

**Comment by MakeSureDudeDies** [comment]
There could such thing be done in pascal (if there is like SDL2 for pascal, but i doubt there is)

**Comment by wdcossey** [comment]
> There could such thing be done in pascal (if there is like SDL2 for pascal, but i doubt there is)

https://github.com/ev1313/Pascal-SDL-2-Headers

https://github.com/PascalGameDevelopment/SDL2-for-Pascal

There's definitely bindings for pascal.
Not sure what SDL2 gives you (GUI wise) though, fpPS4 is Windows only
